tests/modules/commandfilter.c

```diff
@@ -106,6 +122,11 @@ void CommandFilter_CommandFilter(RedisModuleCommandFilterCtx *filter)
             RedisModule_CommandFilterArgInsert(filter, pos + 1,
                     RedisModule_CreateString(NULL, "--inserted-after--", 18));
             pos++;
+        } else if (arg_len == 7 && !memcmp(arg_str, "@retain", 7)) {
+            if (retained) RedisModule_FreeString(NULL, retained);
+            retained = RedisModule_CommandFilterArgGet(filter, pos + 1);
```

Comment:
i think we need to free it in the `RedisModule_OnUnload` function.

Reply:
Fixed.